open-craft/opencraft
# [SE-2741] Clean up tmp folder if OCIM appserver provisioning fails

This PR cleans up tmp folder if OCIM appserver provisioning fails. These folders are used to store git checkouts and virtual environments in `/tmp` folder. If provisioning fails and the size of `/tmp` is large, `tmpreaper` cannot seem to delete unused files.

**JIRA tickets**: [SE-2741](https://tasks.opencraft.com/browse/SE-2741)

**Testing instructions**:

1. Edit [this line](https://github.com/fferrin/opencraft/blob/6451d86484b206739304cd360ee345ba597ce6bd/instance/models/mixins/ansible.py#L141-L147) to raise an `Error`.
1. From any other branch other than this one, launch new appservers and wait until it fails. Check that a folder `tmp <8 chars or number>` appears in the `/tmp` folder with datetime after launching the appserver. 
2. Do the same with this branch and verify that the temporary folders are deleted.

**Reviewers**
